sdk/schema/plugin.go

```diff
@@ -90,6 +90,12 @@ func (p Plugin) Validate() (bool, ValidationReport) {
 		Severity:    ValidationSeverityError,
 	})
 
+	report.AddCheck(ValidationCheck{
+		Description: "Do not specify plugin name in credential usage because credential is from the same plugin",
+		Assertion:   !UnnecessaryPluginNameDefined(p),
+		Severity:    ValidationSeverityWarning,
+	})
+
```

Comment:
I am not really a fan of this, to be honest.

It imposes an artificial restriction on the contributors which does not really improve the code quality/readability (I would even argue that in the case of a common credential name - e.g. Personal Access Token, adding the plugin name too even improves readability - but this is a personal opinion).

I think there are two other problems related to the same thing that are more important to address: 
* the " Credentials referenced in executables are included in the same plugin definition" check, which would fail if any cross-referencing occurs
* enforcing that the "Plugin" value of a "CredentialUsage" actually refers to a plugin. Right now, any value could be in there and validation doesn't trigger anything.

Would love to hear your thoughts @AndyTitu @arunsathiya

Reply:
I already see you captured my second concern in an issue @arunsathiya 😄 https://github.com/1Password/shell-plugins/issues/267

Thanks for the proactive action here!

Reply:
I see this comment is more related to the purpose of the issue we are trying to address itself than the actual implementation. 

This check originated from our Terraform changes in `op`. We are using a credential usage id now to identify credential usages inside an executable. This usage id is stored inside the json file. For old json configurations that don't have a usage id stored in them we need to come up with a default value ourselves, for backwards compatibility and we chose to use the `Name` of the credential usage (because that's what all plugins up to this point have defined). However, if someone were to add a `Plugin` to say the AWS credential usage, that rule would break. So we came up with the enforcement: if a credential usage references the credential in the same plugin, you can't specify the `Plugin`. Otherwise, you must specify the plugin. This explicit separation should help avoid issues/confusion in the future.

Reply:
> the " Credentials referenced in executables are included in the same plugin definition" check, which would fail if any cross-referencing occurs

I do agree we will run into this soon. But for the scope of this PR, there is no immediate need to remove that checks, as no definitions explicitly refer to other plugins.

Reply:
> I see this comment is more related to the purpose of the issue we are trying to address itself than the actual implementation.

Indeed, I preferred leaving a comment here to block the merging of this PR before we address my concern, rather than leave it on the ticket where it could easily be missed. My apologies if this caused any confusion 😄 

> I added it now.

Thank you!

Despite your explanation, I still disagree this is the way to proceed. Such changes should not have collateral impact affecting the user (in this case, contributor) experience. Rather than making assumptions about plugin definitions only enforced by a validation check, we could simply use both keys to look up credentials:
* for credentials usages within the plugin where the credential itself is defined, check both for <credential name> and <plugin name + credential name> (if the `Plugin` field is set).
* for credentials usages within plugins different than the one where the credential is defined, only check for <plugin name + credential name>, enforcing that the plugin name is defined.

Now, I don't feel particularly strong about this, so in the interest of moving forward, I'm okay with using this approach, now that I have the context behind it. However, I would still recommend that we take on the effort of ensuring this is backwards compatible ourselves, without any visible side-effects.